miner/ordering.go

```diff
@@ -36,17 +37,30 @@ type txWithMinerFee struct {
 // newTxWithMinerFee creates a wrapped transaction, calculating the effective
 // miner gasTipCap if a base fee is provided.
 // Returns error in case of a negative effective miner gasTipCap.
-func newTxWithMinerFee(tx *txpool.LazyTransaction, from common.Address, baseFee *uint256.Int) (*txWithMinerFee, error) {
+func newTxWithMinerFee(tx *txpool.LazyTransaction, from common.Address, baseFee *uint256.Int, rates common.ExchangeRates) (*txWithMinerFee, error) {
 	tip := new(uint256.Int).Set(tx.GasTipCap)
 	if baseFee != nil {
-		if tx.GasFeeCap.Cmp(baseFee) < 0 {
+		baseFeeConverted := baseFee
+		if tx.FeeCurrency != nil {
+			baseFeeBig, _ := exchange.ConvertCeloToCurrency(rates, tx.FeeCurrency, baseFee.ToBig())
```

Comment:
We could check for error and return the error here and in L60.

Reply:
We should. If there is an error, `baseFeeBig` will be `nil` which will cause a panic in `Cmp`. The error case is handled sanely everywhere where `newTxWithMinerFee`. I'm not totally sure what result we ideally want (stop precessing tx for now vs removing from tx pool) in every case, but the default error handling should be good enough for such a failure case.
I'll push one more commit to handle the error for both `ConvertCeloToCurrency` calls.

Reply:
Looks like the tx wouldn't be added to the liust or shifted in the first place. So I guess it would be safe to do that.

Reply:
Done. No other changes in the force-push, except for an autosquash to clean up the history now that the review is done.